crates/iceberg/src/arrow/reader.rs

```diff
@@ -186,4 +216,399 @@ impl ArrowReader {
             Ok(ProjectionMask::leaves(parquet_schema, indices))
         }
     }
+
+    fn get_row_filter(&self, parquet_schema: &SchemaDescriptor) -> Result<Option<RowFilter>> {
+        if let Some(predicates) = &self.predicates {
+            let field_id_map = self.build_field_id_map(parquet_schema)?;
+
+            // Collect Parquet column indices from field ids
+            let mut collector = CollectFieldIdVisitor { field_ids: vec![] };
+            visit_predicate(&mut collector, predicates).unwrap();
+            let column_indices = collector
+                .field_ids
+                .iter()
+                .map(|field_id| {
+                    field_id_map.get(field_id).cloned().ok_or_else(|| {
```

Comment:
This doesn't have to be an error? Missing some field maybe valid due to schema evolution.

Reply:
Hmm, if the predicate uses a field not in schema, even with schema evaluation, shouldn't it be considered an error?

E.g., given a predicate `a == 1`, but there is no `a` column.

Reply:
I think this is possible. For example, in version 1 iceberg table has no column a, and we inserted some data. Then we upgraded the schema to add column a, and it becomes version 2. Since iceberg is lazy, parquets files in version 1 has no column a.

Reply:
Okay. Removed the error and just ignored it.